### PR TITLE
Google Books service done & test - Spotify/TMDB service updated

### PR DESCRIPTION
**Pull request by nitek29**

### CODE REVIEW

File: .env.default [hunk 1]
```diff
@@ -25,3 +25,4 @@ SPOTIFY_CLIENT_SECRET=<SPOTIFY_CLIENT_SECRET>
 
 TMDB_CLIENT_TOKEN=<TMDB_CLIENT_TOKEN>
 
+PATH_TO_GOOGLE_OAUTH_FILE=<PATH_TO_GOOGLE_OAUTH_FILE>
```
Comment (RomainCtl) [inline .env.default:28]:
```suggestion
PATH_TO_GOOGLE_OAUTH_FILE=<PATH_TO_GOOGLE_OAUTH_FILE>

```

File: Pipfile [hunk 2]
```diff
@@ -21,10 +21,14 @@ pytest-flask = "*"
 flask-restx = "*"
 flask-bcrypt = "*"
 flask-jwt-extended = "*"
-psycopg2= "*"
+psycopg2-binary= "*"
```
Comment (RomainCtl) [inline Pipfile]:
```suggestion
psycopg2= "*"
```

File: settings/__init__.py [hunk 3]
```diff
@@ -44,3 +44,9 @@
 TMDB_URL_TOKEN = "https://api.themoviedb.org/3/authentication/session/new?api_key="
 TMDB_IMG_URL = 'https://image.tmdb.org/t/p/w500/'
 TMDB_USER_URL = 'https://api.themoviedb.org/3/'
+
+# GOOGLE BOOKS
+GOOGLE_OAUTH_FILE = os.environ['PATH_TO_GOOGLE_OAUTH_FILE']
+GBOOKS_SCOPES = ['https://www.googleapis.com/auth/books']
+GBOOKS_REDIRECT_URL = "http://localhost:4040/api/external/gbook/callback"
+GBOOKS_TOKEN_URI = "https://oauth2.googleapis.com/token"
```
Comment (RomainCtl) [inline settings/__init__.py]:
```suggestion
GBOOKS_TOKEN_URI = "https://oauth2.googleapis.com/token"

```

File: src/__init__.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from flask import Flask
+import os
```
Comment (RomainCtl) [inline src/__init__.py]:
```suggestion
```

File: settings/__init__.py [hunk 5]
```diff
@@ -44,3 +44,9 @@
 TMDB_URL_TOKEN = "https://api.themoviedb.org/3/authentication/session/new?api_key="
 TMDB_IMG_URL = 'https://image.tmdb.org/t/p/w500/'
 TMDB_USER_URL = 'https://api.themoviedb.org/3/'
+
+# GOOGLE BOOKS
+GOOGLE_OAUTH_FILE = os.environ['PATH_TO_GOOGLE_OAUTH_FILE']
+GBOOKS_SCOPES = ['https://www.googleapis.com/auth/books']
+GBOOKS_REDIRECT_URL = "http://localhost:4040/api/external/gbook/callback"
```
Comment (RomainCtl) [inline settings/__init__.py]:
```suggestion
GBOOKS_REDIRECT_URL = URL_FRONT+'/recofinement/user/profile'
```

File: src/model/book_model.py [hunk 6]
```diff
@@ -26,9 +26,9 @@ class BookModel(db.Model):
     author = db.Column(db.String(255), index=True)
     year_of_publication = db.Column(db.Integer)
     publisher = db.Column(db.String(255))
-    image_url_s = db.Column(db.String(255))
-    image_url_m = db.Column(db.String(255))
-    image_url_l = db.Column(db.String(255))
+    image_url_s = db.Column(db.Text)
+    image_url_m = db.Column(db.Text)
+    image_url_l = db.Column(db.Text)
```
Comment (RomainCtl) [inline src/model/book_model.py:31]: if you change column type, you may make a db migration

File: src/service/external_service.py [hunk 7]
```diff
@@ -86,35 +89,34 @@ def get_spotify_data(user_uuid, app):
...
                                 track_id=track.track_id,
                                 play_count=1,
-                                last_played_date=dateutil.parser.isoparse(
-                                    line['played_at'][:-1]) if "played_at" in line.keys() else None
+                                last_played_date=line['played_at']
                             )
                             db.session.add(new_meta_user_track)
                             db.session.flush()
                         else :
                             # ? a verif
```
Comment (RomainCtl) [inline src/service/external_service.py]: checked ?

**Review by RomainCtl — CHANGES_REQUESTED** [review]

File: src/service/external_service.py (hunk 7 above)
Comment (nitek29) [reply src/service/external_service.py]: If speak about the comment, I forgot to remove it but it works

Comment (RomainCtl) [inline src/service/external_service.py]:
```suggestion
```

**Review by RomainCtl — CHANGES_REQUESTED** [review]

**Review by RomainCtl — APPROVED** [review]

**Pull request merged**